Sources/Utility/ArgumentParser.swift

```diff
@@ -277,6 +277,8 @@ protocol ArgumentProtocol: Hashable {
     /// The shell completions to offer as values for this argument.
     var completion: ShellCompletion { get }
 
+    var wasParsed: Bool { get }
```

Comment:
Hi, when I looked through the code which processes the passed arguments, I found that `PostitionalArgument` and `OptionArgument` are handled quite differently. `OptionArgument`s are handled as a map of values where as `PositionalArgument`s are placed into an array and removed when found so that validation for required is simply a matter of checking any left over arguments to see if they are required. 

Because of this I thought it would be simpler in the long run to add a flag to arguments which is set when they are found in the command's arguments. Otherwise I'd have to modify the processing loop in someway and it looked like a far more complex solution. Plus have a flag on arguments seemed like something that might be useful.

Do you have some ideas on an alternative solution?